components/script/dom/htmlmediaelement.rs

```diff
@@ -1220,6 +1233,74 @@ impl HTMLMediaElement {
             },
             PlayerEvent::MetadataUpdated(ref metadata) => {
```

Comment:
On the Kind attribute:
On determining the kind it seems there may not be enough information but I could very well be wrong here as well.

Metadata.format is showing the audio/video type such as 'ogg' where as metadata.audio_tracks / video_tracks shows the specific type such as 'vorbis' for ogg audio. When trying to add the kind to a track it is more specific such as audio/alternate or audio/audiodesc for ogg.

If I am reading this correctly, in the specs that are referenced for the ogg format: [skeleton headers]( https://wiki.xiph.org/SkeletonHeaders),  this would work as long as the tracks are in the correct order. In this case, the first Vorbis audio would be the main audio or main-desc. However, there is also a Vorbis audio for audio descriptions. I wonder if there is only one type of audio for Vorbis would it just be assumed to be the main audio.

DASH seems to be the same way. The spec specifies a role scheme in the form of: urn:mpeg dash:role 2011
This would also require knowing the role type as well in the same metadata.

AC3 and MPEG-2 require bsmod and full_svc values to determine the the correct kind. I haven't fully looked into these though.  

I'm realizing now that Servo may not support all of these as of yet. Is there only specific formats currently?

Reply:
For what it's worth, I looked at Firefox's source code and they only ever create tracks with the "main" type as far as I can tell. I don't think we need to try to figure out specifics here until we come across specific use cases where it matters.

Reply:
Yep you are right. I looked at the source code for Firefox before and could never find it then went back and looked after you mentioned it, and there is a 3 year old open bug for multiple audio/video tracks.

Sounds good, I will finish up what can me implemented and send another push. Thanks.